packages/components/src/engine/metadata.ts

```diff
@@ -60,9 +60,19 @@ const getMetaImage = (props: IsomerPageSchemaType) => {
 
 export const getMetadata = (props: IsomerPageSchemaType) => {
   const faviconUrl = `${props.site.assetsBaseUrl ?? ""}${props.site.favicon || "/favicon.ico"}`
-  const canonicalUrl = props.site.url
-    ? new URL(props.page.permalink, props.site.url).toString()
-    : props.page.permalink
+  const canonicalUrl = (() => {
+    if (!props.site.url) return props.page.permalink
+    try {
+      const siteUrl =
+        props.site.url.startsWith("http://") ||
+        props.site.url.startsWith("https://")
+          ? props.site.url
+          : `https://${props.site.url}`
+      return new URL(props.page.permalink, siteUrl).toString()
+    } catch {
+      return props.page.permalink
+    }
+  })()
```

Comment:
actually i disagree with this approach; i'd rather we just throw an error here so that we can guarantee that the `site.url` in our db is a well-formed url. 

if we go with this approach, we should also disallow `http://`

Reply:
I think @seaerchin  has a good point, we should rather fix and be opinionated about the expected input as the schema is the contract to be followed/adhered to. When it does not match the expected input, it should error out

Else, we will end up with schemas of site that vary between each other.

As per URL standards it should be https:// or http://. But for Isomer, we will strictly disallow http, so it should throw as well for http

Reply:
okay will update!